dartsim/src/SDFFeatures.cc

```diff
@@ -315,8 +323,46 @@ static ShapeAndTransform ConstructGeometry(
 {
   if (_geometry.BoxShape())
     return ConstructBox(*_geometry.BoxShape());
+  else if (_geometry.CapsuleShape())
+  {
+    // TODO(anyone): Replace this code when Capsule is supported by DART
+    common::MeshManager *meshMgr = common::MeshManager::Instance();
+    std::string capsuleMeshName = std::string("capsule_mesh")
+      + "_" + std::to_string(_geometry.CapsuleShape()->Radius())
+      + "_" + std::to_string(_geometry.CapsuleShape()->Length());
+    meshMgr->CreateCapsule(
+      capsuleMeshName,
+      _geometry.CapsuleShape()->Radius(),
+      _geometry.CapsuleShape()->Length(),
+      6, 6);
+    const ignition::common::Mesh * _mesh =
+      meshMgr->MeshByName(capsuleMeshName);
+
+    auto mesh = std::make_shared<CustomMeshShape>(*_mesh, Vector3d(1, 1, 1));
+    auto mesh2 = std::dynamic_pointer_cast<dart::dynamics::MeshShape>(mesh);
+    return {mesh2};
+  }
   else if (_geometry.CylinderShape())
     return ConstructCylinder(*_geometry.CylinderShape());
+  else if (_geometry.EllipsoidShape())
+  {
+    // TODO(anyone): Replace this code when Ellipsoid is supported by DART
```

Comment:
Similar to @azeey's comment about capsule support in DART, are ellipsoids supported in DART? https://dartsim.github.io/dart/v6.9.5/db/da3/classdart_1_1dynamics_1_1EllipsoidShape.html